### ISSUE TITLE
While Loops not working from 24 Apr.

### ISSUE THREAD

**Issue by gobassky** [thunkable/thunkable-issues#493]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

The more info, the faster we will be able to address it!

### Platform
Thunkable ✕, ✕ Android companion and ✕ iOS companion.

### Issue/Request Origin
Own app.

### Steps to Reproduce
1. Any buttons or Functions that contain While loop block will hang both the Andriod and iOS companion
2. No matter how I use the While loop block, as long as it is inside the app process, the app will hang 

### Project Link
https://x.thunkable.com/copy/0cccd3c468bef7c1ee50bd450fc51b2d

### Expected Behavior
(Before 24 Apr), upon pressing the button, the web API will fetch the current date and time. finished the process and use the figure to get the UNIX time. and display at the respective labels.

### Actual Behavior
The app will just hang, the button will be greyed out and the companion app needs to be restarted. The app will only work if all the while loop blocks are removed. 

![while loops](https://user-images.githubusercontent.com/61642929/80676005-63d26500-8ae8-11ea-8d58-edae628d729e.jpg)

**Comment by josmas** [comment]
Hi @gobassky any chance you have an outdated companion?
A simple while app as in here works fine for me: https://x.thunkable.com/copy/547545895f40ef0920ef75d1be1d274c

**Comment by gobassky** [comment]
Thanks @josmas.  I last checked. My companion apps were updated as of 24 Apr which I believe is the latest.  Your "while loop" app indeed is working well on my machine as well. That led me to investigate more what has happened to my program. I realised that when the while loop block is empty or the control conditions of the while loop control is not included in the loop itself, it will hang.  

Perhaps it's my wrong use of while loops to make asynchronous blocks to behave sequentially (such as a series of Web API calls, each must complete before the next). I am not sure what has changed but these were working before, now it is not.  

The error is replicated here:
https://x.thunkable.com/copy/ea0629f118223e828ea37872abe07b81
